### PR TITLE
Fix "guide" notebooks to run under colab

### PR DESCRIPTION
**Pull request by mpharrigan**
- fix a couple of broken links
 - stop using a couple of deprecated cirq functions
 - fixup the dev_tools script that converts these notebooks to recirq.readout_scan.tasks
 - **main**: Upload the guide sample data to figshare. Add utility function to fetch it so notebooks are self-contained

https://figshare.com/articles/dataset/Readout_Scan_Tutorial_Data/13262873

### CODE REVIEW

File: docs/guide/data_analysis.ipynb
Comment (lamberta) [inline docs/guide/data_analysis.ipynb:96]: Getting an install error when I run in Colab: https://colab.research.google.com/github/mpharrigan/ReCirq/blob/2020-11-docs-1/docs/guide/data_collection.ipynb

```
ERROR: pytket 0.6.1 has requirement sympy~=1.5, but you'll have sympy 1.1.1 which is incompatible.
ERROR: pytket-cirq 0.5.0 has requirement cirq~=0.8.0, but you'll have cirq 0.9.1 which is incompatible.
```

The Colab environment might default to a different version than you need. Or maybe pin a version when pip installing?

Comment (mpharrigan) [reply docs/guide/data_analysis.ipynb:96]: Yeah, it says `ERROR` but it should still work. The message is a result of overzealous pinning in cirq *and* pytket

Comment (mpharrigan) [reply docs/guide/data_analysis.ipynb:96]: xref #61 and https://github.com/CQCL/pytket/issues/22 -- which I fixed but looks like we need to bump again. It doesn't help that pytket is closed source so they have to make the change

File: docs/guide/data_analysis.ipynb
Comment (lamberta) [inline docs/guide/data_analysis.ipynb:143]: Got this error in Colab:

```
AttributeError: module 'recirq' has no attribute 'fetch_guide_data_collection_data'
```

Comment (mpharrigan) [reply docs/guide/data_analysis.ipynb:143]: That function is added in this PR, so you can't really test it until this is merged

Comment (lamberta) [reply docs/guide/data_analysis.ipynb:143]: Gotcha. LGTM and we can test once it lands. Thanks

**Review by lamberta — APPROVED** [review]

**Comment by mpharrigan** [comment]
Just verified it runs in colab after merge

**Pull request merged**
